src/hooks/useAssetCaps.tsx

```diff
@@ -0,0 +1,156 @@
+import { useContext, createContext, ReactNode } from 'react';
+import { valueToBigNumber } from '@aave/math-utils';
+import { SupplyCapWarning } from 'src/components/transactions/Warnings/SupplyCapWarning';
+import { BorrowCapWarning } from 'src/components/transactions/Warnings/BorrowCapWarning';
+import { DebtCeilingWarning } from 'src/components/transactions/Warnings/DebtCeilingWarning';
+import { ComputedReserveData } from './app-data-provider/useAppDataProvider';
+import { SupplyCapTooltip } from 'src/components/infoTooltips/SupplyCapTooltip';
+import { BorrowCapTooltip } from 'src/components/infoTooltips/BorrowCapTooltip';
+import { DebtCeilingTooltip } from 'src/components/infoTooltips/DebtCeilingTooltip';
+
+type WarningDisplayProps = {
+  supplyCap?: AssetCapData;
+  borrowCap?: AssetCapData;
+  debtCeiling?: AssetCapData;
+  icon?: boolean;
+  useDefaultTooltip?: boolean;
+};
+
+export type AssetCapData = {
+  percentUsed: number;
+  isMaxed: boolean;
+};
+
+export type AssetCapHookData = AssetCapData & {
+  determineWarningDisplay: (props: WarningDisplayProps) => JSX.Element | null;
+  determineTooltipDisplay: (props: WarningDisplayProps) => JSX.Element | null;
+};
```

Comment:
Some additional fields I think that could help DRY things up relating to these caps are:

- Status color to be used for styling the colors - based off percentUsed
- Tooltip verbiage for good/warning/max
- Warning verbiage for good/warning/max

I think we continue to allocate the logic for displaying these warnings/tooltips to the component that is implementing them.  But we could clean up some of repetitiveness of the tooltip, warning, and status components themselves by abstracting the common stuff to this context.  If you notice, the verbiage is nearly identical, the colors are, and this could be another improvement to this pattern.

```js
type AssetCapWarningTooltipVerbiage = {
   safe: string;
   warning: string;
   maxed: string;
 }
 
type AssetCapStatusColor = "success.main" | "warning.main" | "error.main"

export type AssetCapHookData = AssetCapData & {
  determineWarningDisplay: (props: WarningDisplayProps) => JSX.Element | null;
  determineTooltipDisplay: (props: WarningDisplayProps) => JSX.Element | null;
  tooltipVerbiage: AssetCapWarningTooltipVerbiage;
  warningVerbiage: AssetCapWarningTooltipVerbiage;
  statusColor: AssetCapStatusColor;
```

Then the logic that is in the tooltip components and colors on the status components becomes much more simpler, and maybe the tooltip/warning components become a singular component.

Thoughts on this addition?